File: scripts/vml-abi-probe/langs/drift.cs
// 栈漂移探针（C#）：GenerateConsoleWriteLine 那条路「压了不清」（另一条 stdlibLabel 路
// 本来就由调用方清，两条路约定此前互相矛盾）。判据：`DRIFT=190`（0+1+…+19）。
class Drift
{
    static void Main()
    {
        int s = 0;
        for (int i = 0; i < 20; i++) { System.Console.Write("."); s = s + i; }
        System.Console.Write("\nDRIFT=");
        System.Console.WriteLine(s);
    }
}

File: scripts/vml-abi-probe/langs/drift.r
# 栈漂移探针（R）：print/cat 那条路径「压一个实参」，而被调方已改成裸 `ret` ——
# 压了必须由调用方清（2026-09-17 已补 `ADD R13 #4`）。
#
# 判据：`DRIFT=7`（循环 c(3,4) 两轮都要跑到）。3 次 print 先制造 12 字节漂移，
# 若漂移爬进循环游标，循环就会只跑一轮 ⇒ 得 3。
print(1)
print(2)
print(3)
s <- 0
for (j in c(3, 4)) {
    s <- s + j
}
cat("\nDRIFT=")
print(s)
cat("\n")
